### PR TITLE
refactor(storage): remove compaction group client

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Since we support splitting sstable into different compaction group in meta, the compaction group information of local compaction does not have to be accurate. Besides, the read and write of hummock also does not depend on compaction group id. Therefore, compaction group client is not used in CN and we can remove it. 

In local compaction, we will still group the data by their compaction group id. The compaction group id is obtained by the mapping in hummock version. If the table id of a data is not present in the hummock version, we will assign a default compaction group id for it in local compaction.

Code of notification are changed accordingly. Compaction group change is not notified to CN hummock anymore.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/storage/src/hummock/compactor/shared_buffer_compact.rs [hunk 1]
```diff
@@ -40,14 +42,20 @@ use crate::monitor::StoreLocalStatistic;
...
     let mut grouped_payload: HashMap<CompactionGroupId, UploadTaskPayload> = HashMap::new();
     for uncommitted_list in payload {
         let mut next_inner = HashSet::new();
         for uncommitted in uncommitted_list {
             let compaction_group_id = match &uncommitted {
                 UncommittedData::Sst((compaction_group_id, _)) => *compaction_group_id,
-                UncommittedData::Batch(batch) => batch.compaction_group_id(),
+                UncommittedData::Batch(batch) => {
+                    match compaction_group_index.get(&batch.table_id) {
+                        None => StaticCompactionGroupId::StateDefault as CompactionGroupId,
```
Comment (Li0k) [inline src/storage/src/hummock/compactor/shared_buffer_compact.rs:55]: we need some comment to explain the behavior of server about how to handler the StaticCompactionGroupId ?

**Review by Li0k — APPROVED** [review]
Rest LGTM, PTAL @zwang28

**Review by zwang28 — APPROVED** [review]

File: src/storage/src/hummock/mod.rs [hunk 2]
```diff
@@ -295,8 +269,8 @@ impl HummockStorage {
     }
 
     #[cfg(any(test, feature = "test"))]
```
Comment (zwang28) [inline src/storage/src/hummock/mod.rs:271]: conflict with https://github.com/risingwavelabs/risingwave/pull/5934?

Comment (wenym1) [reply src/storage/src/hummock/mod.rs:271]: #5934 has been reverted in #5937 with a better fix.

**Review by soundOfDestiny — APPROVED** [review]
LGTM. thanks

File: src/storage/src/hummock/compactor/shared_buffer_compact.rs (hunk 1 above)
Comment (soundOfDestiny) [reply src/storage/src/hummock/compactor/shared_buffer_compact.rs:55]: > we need some comment to explain the behavior of server about how to handler the StaticCompactionGroupId ?

The reason here is that
this table is very possible to belong to `StateDefault`

**Pull request merged**